apps/api/src/domain/services/query-services/base-query.service.ts

```diff
@@ -0,0 +1,127 @@
+import {
+    CommandInfo,
+    CommandInfoService,
+    CommandWriteContext,
+} from '../../../app/controllers/command/services/command-info-service';
+import mixTagsIntoViewModel from '../../../app/controllers/utilities/mixTagsIntoViewModel';
+import { InternalError, isInternalError } from '../../../lib/errors/InternalError';
+import { Maybe } from '../../../lib/types/maybe';
+import { isNotFound, NotFound } from '../../../lib/types/not-found';
+import { ResultOrError } from '../../../types/ResultOrError';
+import { TagViewModel } from '../../../view-models/buildViewModelForResource/viewModels';
+import { BaseViewModel } from '../../../view-models/buildViewModelForResource/viewModels/base.view-model';
+import { Resource } from '../../models/resource.entity';
+import { Song } from '../../models/song/song.entity';
+import { ISpecification } from '../../repositories/interfaces/ISpecification';
+import { AggregateId, isAggregateId } from '../../types/AggregateId';
+import { InMemorySnapshot, ResourceType } from '../../types/ResourceType';
+import { GeneralQueryOptions } from './types/GeneralQueryOptions';
+import getDefaultQueryOptions from './utilities/getDefaultQueryOptions';
+
+type ResourceByIdQueryResult<UViewModel extends BaseViewModel> = {
+    data: UViewModel;
+    actions: CommandInfo[];
+};
+
+type ResourceIndexQueryResult<UViewModel extends BaseViewModel> = {
+    data: ResourceByIdQueryResult<UViewModel>[];
+    actions: CommandInfo[];
+};
+
+type ViewModelWithTags<T> = T & { tags: TagViewModel[] };
+
+export abstract class BaseQueryService<
+    TDomainModel extends Resource,
+    UViewModel extends BaseViewModel
+> {
+    constructor(
+        private readonly type: ResourceType,
+        protected readonly commandInfoService: CommandInfoService
```

Comment:
protected can still be accessed outside this abstract class?

Reply:
Only by one of its descendant classes.